src/utils/SmtSolver.cc

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023, Martin Blicha <martin.blicha@gmail.com>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "SmtSolver.h"
+
+SMTSolver::SMTSolver(Logic & logic, WitnessProduction setup) {
```

Comment:
Maybe also make config as a part of constructor?  What if we need engine-specific configuration?

Reply:
You can get the config with `getConfig` method and set what you need there.
Requiring `SMTConfig` in the constructor would defeat the purpose of the proposed change.

Later, I imagine we identify what we require and have methods for that, so that we can hide `MainSolver` and `SMTConfig` completely.

Reply:
Gotcha